aten/src/ATen/native/cpu/ReduceOpsKernel.cpp

```diff
@@ -112,12 +112,65 @@ static void cumprod_cpu_kernel(const Tensor& result, const Tensor& self, int64_t
     );
   });
 }
+// custom min and max to be used in logcumsumexp for complex arguments
+template <typename scalar_t>
+c10::complex<scalar_t> _logcumsumexp_minmax(c10::complex<scalar_t> x, c10::complex<scalar_t> y, bool min) {
```

Comment:
Perf consideration: Passing `bool` as template argument would make the code much faster (as the last statement can be evaluated at compile time rather than at runtime
```suggestion
template <typename scalar_t, bool min>
c10::complex<scalar_t> _logcumsumexp_minmax(c10::complex<scalar_t> x, c10::complex<scalar_t> y) {
```

Reply:
Done in the next commit